src/reporters.ts

```diff
@@ -26,15 +26,70 @@ const getContextPath = (context: Context): string => {
   return `${keysPath}] is not a valid [${lastType.name}]`;
 };
 
+/**
+format path without usless indexes
+ */
+const isArrayIndex = (
+  prev: ReadonlyArray<string>,
+  c: ContextEntry,
+  i: number,
+  context: Context
+): ReadonlyArray<string> =>
+  c.key !== "" &&
+  ((Number.isInteger(+c.key) &&
```

Comment:
may you please add a comment on what are we checking here? I'm asking since
the first branch checks for `Number.isInteger && ArrayType.is` while the second part checks only if `ReadonlyArrayType.is` (without checking the type of contained items).

(maybe there's a misplaced parens?)